src/main.cpp

```diff
@@ -1280,17 +1340,28 @@ bool CheckTransactionWithoutProofVerification(const CTransaction& tx, CValidatio
                             REJECT_INVALID, "bad-tx-expiry-height-too-high");
         }
     }
+    auto orchard_bundle = tx.GetOrchardBundle();
 
     // Transactions containing empty `vin` must have either non-empty
     // `vJoinSplit` or non-empty `vShieldedSpend`.
-    if (tx.vin.empty() && tx.vJoinSplit.empty() && tx.vShieldedSpend.empty())
+    if (tx.vin.empty() &&
+        tx.vJoinSplit.empty() &&
+        tx.vShieldedSpend.empty() &&
+        !orchard_bundle.SpendsEnabled())
```

Comment:
I guess these checks are really just defense-in-depth, because they don't really guarantee anything other than a structural validity property. IMO that's more a part of parsing than an important consensus rule; zero-valued notes are possible, right? So this is just a fail-fast filter before we go to check the value balance?

Reply:
Correct, this is a fail-fast check that rejects obviously-invalid transaction constructions that the parser doesn't structurally prevent.

Reply:
This doesn't seem to implement the specified consensus rule, because it doesn't check `nActionsOrchard`.